From: Felix Lechner <felix.lechner@lease-up.com>
Date: Sat, 2 Sep 2017 09:57:43 -0300
Subject: Remove race conditions from tests

Testing the qmqp protocol fails because the testing server cannot
acquire the lock.  This is solved by waiting for the previous
server to exit gracefully and clear the lock. Similarly, wait
before removing temporary directories.

Forwarded: no
Last-Update: 2017-10-07
---
 test/functions.in    | 1 +
 test/tests/protocols | 1 +
 2 files changed, 2 insertions(+)

diff --git a/test/functions.in b/test/functions.in
index 0ce02bd..2a8ab8d 100644
--- a/test/functions.in
+++ b/test/functions.in
@@ -20,6 +20,7 @@ exit_cleanup() {
     # Remove all temporary files on success
     if [ $? -eq 0 ]
     then
+        wait
         rm -rf $tmpdir
     fi
 }
diff --git a/test/tests/protocols b/test/tests/protocols
index 49ca530..2678844 100644
--- a/test/tests/protocols
+++ b/test/tests/protocols
@@ -66,6 +66,7 @@ do
 	echo "Testing protocol failure with $p."
 	error 11 protocol $p -p $port --host=localhost 3<testmail
 	stop server
+	wait
 done
 
 rm -f testmail
